### ISSUE TITLE
Upload images to Mosaico Gallery do not work in backdrop

### ISSUE THREAD

**Issue by catorghans** [veda-consulting/uk.co.vedaconsulting.mosaico#511]
Mosaico image upload does not function in Backdrop.
Nothing in civicrm log (or backdrop log), so it gets difficult to debug.

When you upload from witihin a message, it shows a small version during upload, but when it's done, you just get another picture.

**Comment by catorghans** [comment]
What I noticed is that a POST to index.php?q=civicrm/mosaico/upload is redirected to a GET to /civicrm/mosaico/upload
![Schermafbeelding_20221228_220527](https://user-images.githubusercontent.com/961049/209872231-b7d139d0-6409-49f1-8d69-e7202a1f7bb1.png)

**Comment by catorghans** [comment]
OK, if people run into this, I'll describe the solution I found.

The issue was caused by the globalredirect module in Backdrop.
If you have the "Non-clean to Clean" option set, it causes the index.php?q=civicrm/mosaico/upload url to be changed to
/civicrm/mosaico/upload.

Imo it should not do that with POSTS, or at least make it a POST again with every piece of information in it.

But... when you switch this option off, the Mosaico upload works again!
